onnxruntime/core/providers/cpu/ml/ml_common.h

```diff
@@ -258,7 +259,7 @@ static inline float sigmoid_probability(float score, float proba, float probb) {
 }
 
 template <typename T>
-static inline void ComputeSoftmax(const gsl::span<T>& values) {
+static inline void ComputeSoftmax(gsl::span<T>& values) {
```

Comment:
>an<T>& [](http://example.com/codeflow?start=41&length=6)

const T. That's why InlinedVector does not convert automatically to span. Everywhere const ref to a container is passed gsl::span<const T>

Reply:
It did not work with the const. I can check again. I removed the const because the function modifies values inplace. Technically the span is not modified but its content is modified. That's why I removed it. Let me know if that's ok.

Reply:
Okay, then it is really an out parameter. span is actually not that good for this situation.

Reply:
This function is called from different kind of container including span. Changing the input type means changing other lines as well.

Reply:
consider passing span by value, `gsl::span<T> values`
it doesn't need to be by non-const ref as the span itself is not modified, and span should be cheap to copy

Reply:
I tried but the compiler cannot implicitely convert InlinedVector<T> into gsl::span<T> so I let it unchanged.

Reply:
I see, yeah it should be convertible to `span<const T>` but we are modifying the elements.
what I was suggesting here is replacing `span<T>&` with `span<T>`. I think directly passing InlinedVector wouldn't work for either.